controllers/configurationpolicy_controller.go

```diff
@@ -1371,97 +1397,171 @@ func (r *ConfigurationPolicyReconciler) determineDesiredObjects(
 		return nil, &scopedGVR, errEvent, nil
 	}
 
+	// Fetch and filter objects using provided objectSelector
 	desiredObjects := []*unstructured.Unstructured{}
+
+	// The object either has no name specified or it is templated in the object definition
+	if desiredName == "" && objectT.ObjectSelector != nil {
+		selector, err := metav1.LabelSelectorAsSelector(objectSelector)
+		if err != nil {
+			log.Error(err, "Failed to select the resources",
+				"objectSelector", fmt.Sprint(objectT.ObjectSelector.String()))
+
+			msg := fmt.Sprintf("Error parsing provided objectSelector: %v", err)
+
+			// only report this error if there wasn't another yet
+			errEvent := &objectTmplEvalEvent{
+				compliant: false,
+				reason:    "objectSelector error",
+				message:   msg,
+			}
+
+			return nil, &scopedGVR, errEvent, err
+		}
+
+		filteredObjects, err := r.DynamicWatcher.List(plc.ObjectIdentifier(), objGVK, "", selector)
+		if err != nil {
+			log.Error(err, "Failed to fetch the resources",
+				"objectSelector", fmt.Sprint(objectT.ObjectSelector.String()))
+
+			msg := fmt.Sprintf("Error listing resources with provided objectSelector: %v", err)
```

Comment:
Could you please include the object-template index in the error message?

Reply:
Good catch on adding the index to the messages! 😄